presto-spi/src/main/java/com/facebook/presto/spi/function/RoutineCharacteristics.java

```diff
@@ -94,4 +116,45 @@ public String toString()
     {
         return format("(%s, %s, %s)", language, determinism, nullCallClause);
     }
+
+    public static class Builder
+    {
+        private Language language = SQL;
+        private Determinism determinism = NOT_DETERMINISTIC;
+        private NullCallClause nullCallClause = CALLED_ON_NULL_INPUT;
+
+        private Builder()
```

Comment:
I think we normally do empty constructor in a single line
`private Builder() {}`